src/tezos_fmt_plug.c

```diff
@@ -0,0 +1,339 @@
+/*
+ * JtR format to crack password protected Tezos keys.
+ *
+ * This software is Copyright (c) 2018, Dhiru Kholia <dhiru at openwall.com>
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * Special thanks goes to https://github.com/NODESPLIT/tz-brute for making this
+ * work possible.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_tezos;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_tezos);
+#else
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#define OMP_SCALE               1
+
+#include "arch.h"
+#include "misc.h"
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "options.h"
+#include "jumbo.h"
+#include "ed25519.h"
+#include "blake2.h"
+#define PBKDF2_HMAC_SHA512_VARYING_SALT 1
+#include "pbkdf2_hmac_sha512.h"
+#include "memdbg.h"
+
+#define FORMAT_NAME             "Tezos Key"
+#define FORMAT_LABEL            "tezos"
+#define FORMAT_TAG              "$tezos$"
+#define TAG_LENGTH              (sizeof(FORMAT_TAG) - 1)
+#if ARCH_BITS >= 64
+#define ALGORITHM_NAME          "PBKDF2-SHA512 64/" ARCH_BITS_STR " " SHA2_LIB
+#else
+#define ALGORITHM_NAME          "PBKDF2-HMAC-SHA512 32/" ARCH_BITS_STR " " SHA2_LIB
+#endif
```

Comment:
This ended up being wrong. First, @magnumripper corrected me that we're no longer listing "HMAC". Then, this now says "HMAC" on 32-bit (it shouldn't), but not on 64-bit. Finally, it fails to report on use of SIMD (it should).

Reply:
Yep!

Does the following patch look correct?

```diff
diff --git a/src/tezos_fmt_plug.c b/src/tezos_fmt_plug.c
index d1a7dfce5..b7fe681c5 100644
--- a/src/tezos_fmt_plug.c
+++ b/src/tezos_fmt_plug.c
@@ -41,10 +41,14 @@ john_register_one(&fmt_tezos);
 
 #define FORMAT_NAME             "Tezos Key"
 #define FORMAT_LABEL            "tezos"
+#ifdef SIMD_COEF_64
+#define ALGORITHM_NAME          "PBKDF2-SHA512 " SHA512_ALGORITHM_NAME
+#else
 #if ARCH_BITS >= 64
 #define ALGORITHM_NAME          "PBKDF2-SHA512 64/" ARCH_BITS_STR " " SHA2_LIB
 #else
-#define ALGORITHM_NAME          "PBKDF2-HMAC-SHA512 32/" ARCH_BITS_STR " " SHA2_LIB
+#define ALGORITHM_NAME          "PBKDF2-SHA512 32/" ARCH_BITS_STR " " SHA2_LIB
+#endif
 #endif
 #define BENCHMARK_COMMENT       ""
 #define BENCHMARK_LENGTH        0
```

I don't really understand this stuff ;)